scripts/build-circt.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+GIT_COMMIT=093cdfe482530623fea01e1d3242af93e533ba54
+
+CIRCT_BUILD=${PWD}/build-circt
+CIRCT_INSTALL=${CIRCT_BUILD}/circt
+
+function commit()
+{
+	echo ${GIT_COMMIT}
+}
+
+function usage()
+{
+	echo "Usage: ./build-circt.sh [OPTION] [VAR=VALUE]"
+	echo ""
+	echo "  --build-path PATH     The path where to build CIRCT."
+	echo "                        [${CIRCT_BUILD}]"
+	echo "  --install-path PATH   The path where to install CIRCT."
```

Comment:
Could we make these two arguments mandatory for the script and not provide default values for it? Default values makes everything always implicit, potentially leading to nasty surprises.

Reply:
Same as above. There is a default location for where the build happens, and the normal user never has to think about it.

Reply:
As the README.md says, the normal user would simply do
```
./scripts/build-circt.sh
./configure --enable-hls
```
instead of having to specify paths where to place things.

Reply:
I would even go so far as if `--enable-hls` is given and the `build-circt` directory does not exist, then the `./scripts/build-circt.sh` is called under the hood.

Reply:
One option would be to add a make target for building CIRCT that provides the configured path.
That would eliminate the default paths in the build-circt.sh script.```

In the root Makefile the lines starting with .PHONY could be added

```
ifdef HLS_ENABLED
include jlm/hls/Makefile.sub
include tools/jhls/Makefile.sub
include tools/jlm-hls/Makefile.sub

.PHONY: build-circt
build-circt: $(CIRCT_PATH)

$(CIRCT_PATH):
        ./scripts/build-circt.sh --build-path ./build-circt --install-path $(CIRCT_PATH)
endif
```